src/core.js

```diff
@@ -239,9 +240,11 @@ jQuery.extend( {
 			return false;
 		}
 
-		// If the function hasn't returned already, we're confident that
-		// |obj| is a plain object, created by {} or constructed with new Object
-		return true;
+		// Own properties are enumerated firstly, so to speed up,
+		// if last one is own, then all properties are own
+		for ( key in obj ) {}
+
+		return key === undefined || hasOwn.call( obj, key );
```

Comment:
I like this restoration, but would _love_ to see the whole function simplified (_especially_ for 3.0): http://jsbin.com/hezeqilega
```js
isPlainObject: function ( obj ) {
	var proto, Ctor;

	// Catch obvious negatives
	if ( !obj || toString.call( obj ) !== "[object Object]" ) {
		return false;
	}

	// Rely on Object.getPrototypeOf where available
	if ( getPrototypeOf ) {
		proto = getPrototypeOf( obj );

		// Objects with no prototype (e.g., `Object.create( null )`) are plain
		if ( !proto ) {
			return true;
		}

		Ctor = hasOwn.call( proto, "constructor" ) && proto.constructor;

    // Pre-3.0 fallback
	} else {
		Ctor = obj.constructor;
	}

	// All remaining plain objects were constructed by a global Object
	return typeof Ctor === "function" && fnToString.call( Ctor ) === ObjectFunction;
}
```